docs/micropython.rst

```diff
@@ -0,0 +1,91 @@
+MicroPython
+***********
+
+.. py:module:: micropython
+
+Access and control MicroPython internals.
+
+Functions
+=========
+
+.. py:function:: micropython.const(expr)
+
+    Used to declare that the expression is a constant so that the compiler can 
+    optimise it. The use of this function should be as follows:
+
+    .. code-block:: python
+
+        from micropython import const
+        CONST_X = const(123)
+        CONST_Y = const(2 * CONST_X + 1)
+
+    Constants declared this way are still accessible as global variables from 
+    outside the module they are declared in. On the other hand, if a constant 
+    begins with an underscore then it is hidden, it is not available as a 
+    global variable, and does not take up any memory during execution.
+
+
+.. py:function:: micropython.opt_level([level])
+
+    If level is given then this function sets the optimisation level for 
+    subsequent compilation of scripts, and returns None. Otherwise it returns 
+    the current optimisation level.
+
+    The optimisation level controls the following compilation features:
+
+* Assertions: at level 0 assertion statements are enabled and compiled 
+    into the bytecode; at levels 1 and higher assertions are not compiled.
+
+* Built-in ``__debug__`` variable: at level 0 this variable expands to 
+    True; at levels 1 and higher it expands to False.
+
+* Source-code line numbers: at levels 0, 1 and 2 source-code line number 
+    are stored along with the bytecode so that exceptions can report the line 
+    number they occurred at; at levels 3 and higher line numbers are not 
+    stored.
+
+    The default optimisation level is usually level 0.
+
+
+.. py:function:: micropython.mem_info([verbose])
+
+    Print information about currently used memory. If the verbose argument is 
+    given then extra information is printed.
+
+
+.. py:function:: micropython.qstr_info([verbose])
+
+    Print information about currently interned strings. If the verbose argument
+    is given then extra information is printed.
+
+    The information that is printed is implementation dependent, but currently 
```

Comment:
We can remove this first sentence about being implementation dependent, as the following text does describe the micro:bit implementation well. So we can start this paragraph with "It includes the number of interned..."